### PR TITLE
draft version of MVA producer and bjet-regression, global vars from collections

### PR DESCRIPTION
**Pull request by arizzi**
Add generic TMVA producer with variables computed from StringParser
Add bJet regression MVA producer adding variables not computable with StringParser
Add possibility to compute globalvariables from collections (size, max,min, sumpt etc..), and use it for SA

### CODE REVIEW

**Comment by arizzi** [comment]
how do we want to handle secondary vertices? 
the lack of user floats would force us to recompute dLen, dLenSig etc.. in each producer that may needed it (as we do not want to read back from FlatTable!).

Should we define a  
pat::SecondaryVertex : public pat::Object<VertexCompositePtrCandidate>
to support userFloats and  userCands?

File: plugins/BaseMVATemplateProducer.h [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  mvaOut.reserve(src->size());
+  for(auto const & o: *src) {
+	for(auto const & p : funcs_ ){
+		values_[positions_[p.first]]=p.second(o);
+	}
+        fillAdditionalVariables(o);
+	mvaOut.push_back(reader_.EvaluateRegression(collName_+"_"+attName_)[0]);
+  }
+  O dumb;
+  std::unique_ptr<O> mvaV(prepareOutput(src,dumb,mvaOut));
```
Comment (arizzi) [inline plugins/BaseMVATemplateProducer.h]: this is silly and should be fixed

**Comment by gpetruc** [comment]
i don't think there's many places that need the dlen & sig, only the b regression and the flat table producer.
anyway, if there are many, they could be produced as valuemaps and then we implement reading vmaps in the table producer

**Comment by arizzi** [comment]
@gpetruc have a look , I think we can merge

**Pull request merged**